File: go/500.go
import (
    "strings"
    "fmt"
)

func findWords(words []string) []string {
    row1 := "qwertyuiop"
    row2 := "asdfghjkl"
    row3 := "zxcvbnm"

    result := make([]string, 0)

    for _,word := range words {
        fmt.Println(word)
        temp := strings.ToLower(word)
        char1 := string(temp[0])
        fmt.Println(char1)


        addWord := true

        if strings.Contains(row1, string(char1)) {

            for _, char := range temp {
                if !strings.Contains(row1, string(char)) {
                    addWord = false
                    break
                }
            }

        }else if strings.Contains(row2, string(char1)) {

            for _, char := range temp {
                if !strings.Contains(row2, string(char)) {
                    addWord = false
                    break
                }
            }
        }else {

            for _, char := range temp {
                if !strings.Contains(row3, string(char)) {
                    addWord = false
                    break
                }
            }
        }
        if addWord {
            result = append(result, word)
        }
    }
    return result
}

File: py/500.py
# Author: Andrew Hamara

# Solution to leetcode problem 500. Keyboard Row

class Solution:
    def findWords(self, words:List[str]) -> List[str]:
        row1 = {'q', 'w', 'e', 'r', 't', 'y', 'u', 'i', 'o', 'p'}
        row2 = {'a', 's', 'd', 'f', 'g', 'h', 'j', 'k', 'l'}
        row3 = {'z', 'x', 'c', 'v', 'b', 'n', 'm'}

        matches = []
        for x in words:
            if (self.onlyInRow(x, row1) or
                self.onlyInRow(x, row2) or
                self.onlyInRow(x, row3)):
                    matches.append(x)
        return matches

    def onlyInRow(self, s:str, row:set[str]) -> bool:
        for x in s:
            if x.lower() not in row:
                return False
        return True
